### DISCUSSION TITLE
最期待 FlyCV 升级支持的新特性

### DISCUSSION THREAD

**Discussion by oodream** [PaddlePaddle/FlyCV#108] [Ideas]
建议，写下当前最希望得到升级支持的新特性，帮助 您的产品以及FlyCV 变得越来越好！

**Comment by oodream** [comment]
希望支持Python接口

**Comment by aslily1234** [comment]
希望能够支持对视频文件的读写，对视频流的读取。

**Comment by Dandelion111** [comment]
希望能够支持基本的图像处理算法，比如滤波、二值化、增强等，相当于opencv中的imgproc模块

**Comment by sdcb** [comment]
希望提供更多、完整的C API

**Comment by ChaoII** [comment]
希望提供本土化，优化的`putText`方法， 类似于`cv::freetype3::putText`这个接口，目前opencv很大的痛点就是putText 功能垃圾。字体丑，不支持utf8等等

**Comment by sdcb** [comment]
希望也提供x86的Vector相关优化。

**Comment by sdcb** [comment]
希望将PaddlePaddle相关的示例（PaddleOCR, PaddleDetection、PaddleClassification、PaddleSeg）改成使用FlyCV进行。

**Comment by mkwml** [comment]
希望支持float类型的resize函数

**Comment by sdcb** [comment]
希望提供demosaicing函数，它可以将原始图片的拜耳排列格式（RGGB）转换为RGB格式。
OpenCV文档：https://docs.opencv.org/4.x/d8/d01/group__imgproc__color__conversions.html#ga57261f12fccf872a2b2d66daf29d5bd0

**Comment by sdcb** [comment]
不知道这个项目还是否在继续推进，我仔细研究我了的C# [PaddleSharp](https://github.com/sdcb/paddlesharp)项目用到的OpenCV算子和FlyCV的对应关系，发现结论如下，请参考：

以下是 OpenCV 算子与 FlyCV 算子的对应关系：

1. `Mat.Dispose()` --> `fcvReleaseCMat`
2. `Cv2.GetPerspectiveTransform()` --> `fcvGetPerspectiveTransform`
3. `Mat.WarpPerspective()` --> `fcvWarpPerspective`
4. `Cv2.Transpose()` --> `fcvTranspose`
5. `Cv2.Flip()` --> `fcvFlip`
6. `Cv2.GetStructuringElement()` --> 无对应算子
7. `Cv2.Dilate()` --> 无对应算子
8. `Mat.FindContoursAsArray()` --> 无对应算子
9. `Mat.MinMaxIdx()` --> `fcvMinMaxLoc`
10. `Mat.CopyTo()` --> 无对应算子，但可能使用 `fcvMatrixMul` 配合参数实现
11. `Cv2.Rotate()` --> 无对应算子，但可能使用 `fcvGetRotationMatrix2D` 和 `fcvWarpAffine` 组合实现
12. `Mat.Resize()` --> `fcvResize`
13. `Mat.CvtColor()` --> `fcvCvtColor`
14. `Mat.Threshold()` --> 无对应算子
15. `Mat.CopyMakeBorder()` --> `fcvCopyMakeBorder`
16. `Mat.ConvertTo()` --> 无对应算子
17. `Cv2.Split()` --> `fcvSplitToMemcpy`
18. `Cv2.Merge()` --> 无对应算子，但可能使用 `fcvMatrixMul` 配合参数实现
19. `Cv2.ExtractChannel()` --> `fcvExtractChannel`
20. `Mat.DrawContours()` --> `fcvPolyLines`（可能有类似功能，但不完全一样）

以下是无法在 FlyCV 中找到对应算子的 OpenCV 算子：

- `Cv2.GetStructuringElement()`
- `Cv2.Dilate()`
- `Mat.FindContoursAsArray()`
- `Mat.Threshold()`
- `Mat.ConvertTo()`

我相信我的项目和FlyCV最重要的隔阂是下面两个算子：
* Dilate
* FindContoursAsArray
